sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -921,25 +943,20 @@ private long insertRevision(SQLDatabase db,InsertRevisionOptions options) {
             logger.fine("New revision inserted: " + options.docNumericId + ", " + options.revId);
             newSequence = db.insert("revs", args);
             if (newSequence < 0) {
-                throw new IllegalStateException("Unknown error inserting new updated doc, please checking log");
+                throw new IllegalStateException("Unknown error inserting new updated doc, please check log");
```

Comment:
I wish I knew what this referred to. I guess the original author assumed there would always be a log.  I'm not convinced this exception is right either, Do you have a view on what exception should be thrown?